snuba/datasets/factory.py

```diff
@@ -53,12 +54,19 @@ def get_dataset(name: str) -> Dataset:
 
     try:
         dataset = DATASETS_IMPL[name] = dataset_factories[name]()
+        DATASETS_NAME_LOOKUP[dataset] = name
     except KeyError as error:
         raise InvalidDatasetError(f"dataset {name!r} does not exist") from error
 
     return dataset
 
 
+def get_dataset_name(dataset: Dataset) -> str:
+    if dataset not in DATASETS_NAME_LOOKUP:
+        raise InvalidDatasetError("Dataset name not specified")
+    return DATASETS_NAME_LOOKUP[dataset]
```

Comment:
It'd probably be a bit cleaner to write this like `get_dataset` and just raise from the `KeyError` (see line 58-59) so that the full traceback is maintained.